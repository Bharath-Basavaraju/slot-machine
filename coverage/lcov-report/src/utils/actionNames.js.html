<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/utils/actionNames.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/utils</a> actionNames.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span></td><td class="text"><pre class="prettyprint lang-js">// Action names
export let SET_ROLLER_DATA = "SET_ROLLER_DATA";
export let SET_SLOT_ELEMENT =  "SET_SLOT_ELEMENT";
export let WAIT_TO_START_SPIN = "WAIT_TO_START_SPIN";
export let SET_WAIT_TO_SPIN = "SET_WAIT_TO_SPIN";
export let START_SPIN = "START_SPIN";
export let SET_SPINNING = "SET_SPINNING";
export let SPINNING_STARTED = "SPINNING_STARTED";
export let SET_STOP_SPINNING = "SET_STOP_SPINNING";
export let STOP_SPINNING = "STOP_SPINNING";
export let SPINNING_STOPPED = "SPINNING_STOPPED";</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 16 2018 11:16:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
