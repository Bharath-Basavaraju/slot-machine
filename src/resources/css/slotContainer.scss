.slotMachine{
    margin: 0 auto;
    .slotWrapper{
        overflow: hidden;
        height: 100px;
        padding: 2em 0;
        display: flex;
        transition: 0.05s transform;
        &::after {
            content: '';
            display: block;
            height: 100px;
            background: rgba(255, 0, 0, .1);
        }
    }
    .btnContainer{
        clear: both;
        text-align: center;
        .defaultBtn{
            cursor: pointer;
            width: 100px;
            background-color: #45f924;
            font-size: 16px;
            height: 40px;
            text-align: center;
            margin: 10px;
            border: 1px solid #000;
            border-radius: 5px;
            font-weight: bold;
            &.disabled{
                cursor: default;
                background-color: burlywood;
            }
        }
    }
    .resultContainer{
        text-align: center;
        font-size: 35px;
        line-height: 70px;
        color: #F53434;
        text-transform: uppercase;
        font-weight: bold;
        animation: blinkingText 1.5s linear infinite;
    }
}

@keyframes blinkingText{
    0%{opacity: 0;}
    25%{opacity: 0.35;}
    50%{opacity: 0.5;}
    75%{opacity: 0.75;}
    100%{opacity: 1;}
}

