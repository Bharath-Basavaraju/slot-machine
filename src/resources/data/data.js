export const rollerData = [
        {
            id: 'strawberry',
            backgroundPosition: '-0 -0'
        },
        {
            id: 'banana',
            backgroundPosition: '-0 -100px'
        },
        {
            id: 'orange',
            backgroundPosition: '-0 -200px'
        },
        {
            id: 'monkey',
            backgroundPosition: '-0 -300px'
        }
];