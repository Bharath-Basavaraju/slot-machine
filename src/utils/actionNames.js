// Action names
export let SET_ROLLER_DATA = "SET_ROLLER_DATA";
export let SET_SLOT_ELEMENT =  "SET_SLOT_ELEMENT";
export let WAIT_TO_START_SPIN = "WAIT_TO_START_SPIN";
export let SET_WAIT_TO_SPIN = "SET_WAIT_TO_SPIN";
export let START_SPIN = "START_SPIN";
export let SET_SPINNING = "SET_SPINNING";
export let SPINNING_STARTED = "SPINNING_STARTED";
export let SET_STOP_SPINNING = "SET_STOP_SPINNING";
export let STOP_SPINNING = "STOP_SPINNING";
export let SPINNING_STOPPED = "SPINNING_STOPPED";