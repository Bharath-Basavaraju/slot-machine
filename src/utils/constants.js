export const numberOfRoller = 3;
export const numberOfSlotElement = 4;
export const spinSpeed = 50; // in mS
export const defaultStartSpin = 5000; //in ms
export const defaultStopSpin = 10000; // in ms
export const firstPrize = "You Won $100";
export const secondPrize = "You Won $20";
export const thirdPrize = "You Won $10";
export const lostMsg = "Sorry, you lost!";
