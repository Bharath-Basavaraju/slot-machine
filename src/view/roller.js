import React from "react";
import * as slotRowStyles from '../resources/css/roller.scss';

export default class RollerComponent extends React.Component {
    render() {
        let slotData = this.props.elementData;
        return (
            <div className={slotRowStyles.slotRow} style={{backgroundPosition: slotData.backgroundPosition ? slotData.backgroundPosition : "0 0"}}></div>
        );
    }
}