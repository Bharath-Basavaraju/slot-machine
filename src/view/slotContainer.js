import React from 'react';
import { connect } from 'react-redux';
import * as slotContainerStyles from '../resources/css/slotContainer.scss';
import RollerComponent from './roller';
import * as actions from '../actions/slotActions';

export class SlotContainerComponent extends React.Component {
    constructor(props) {
        super(props);
        //this.props.setSlotElements();
        //this.props.waitToStartSpin();
    }

    render() {
        const rollers = [];
        this.props.rollers.forEach((element, index) => {
            rollers.push(
                <RollerComponent key={index} elementData={element} />
            );
        });
        return (
            <div className={slotContainerStyles.slotMachine}>
                <div className={slotContainerStyles.slotWrapper}>
                    { rollers }
                </div>
                <div className={slotContainerStyles.btnContainer}>
                    <button id={"startBtn"}
                        className={slotContainerStyles.defaultBtn +" "+ (this.props.spinning ? slotContainerStyles.disabled : '')}
                        onClick={this.props.onStart}
                        disabled={this.props.spinning}>
                        Start
                    </button>
                    <button id={"stopBtn"}
                        className={slotContainerStyles.defaultBtn +" "+ (!this.props.spinning ? slotContainerStyles.disabled : '')}
                        onClick={this.props.onStop}
                        disabled={!this.props.spinning}>
                        Stop
                    </button>
                </div>
                { this.props.result
                    ?   <div className={slotContainerStyles.resultContainer}>
                            {this.props.result}
                        </div>
                    : null
                }
            </div>
        );
    }
}
export default connect(
    (store, ownProps)=>{
        return{
            slotMachine : store,
            rollers: store.get("rollers"),
            spinning: store.get("spinningFlag"),
            result: store.get("result"),
            ...ownProps
        }
    },
    (dispatch)=>{
        return{
            /*setSlotElements: () => {
              dispatch(actions.setSlotElements());
            },
            waitToStartSpin: () => {
              dispatch(actions.waitToStartSpin());
            },*/
            onStart: () => {
                dispatch(actions.startSpin());
            },
            onStop: () => {
                dispatch(actions.stopSpinning());
            }
        }
    }
)(SlotContainerComponent);