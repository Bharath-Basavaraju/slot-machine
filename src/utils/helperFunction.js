import * as constants from './constants';

export function generateRandomSlotItem() {
    return Math.floor(Math.random() * ( constants.numberOfSlotElement - 1 + 1)) + 1;
}

export function getResult(rollers){
    if(rollers[0].id === rollers[1].id && rollers[1].name === rollers[2].id){
        return constants.firstPrize;
    }
    else if(rollers[0].id === rollers[1].id || rollers[1].name === rollers[2].id){
        return constants.secondPrize;
    }
    else if(rollers[0].id === rollers[2].id){
        return constants.thirdPrize;
    }
    else{
        return constants.lostMsg;
    }
}

