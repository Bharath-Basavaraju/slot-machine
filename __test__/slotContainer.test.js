import {SlotContainerComponent} from "../src/view/slotContainer"
import React from 'react';
import {shallow} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { rollerData } from '../src/resources/data/data';
import * as constants from "../src/utils/constants";
// // Create the mock store
// import configureMockStore from 'redux-mock-store';
// const mockStore = configureMockStore();

function setup(props){
    const wrapper = shallow(<SlotContainerComponent {...props} />);
    return {
        props,
        wrapper
    }
}

const mockOnStartSpin = jest.fn();
const mockOnStopSpin = jest.fn();

function generateRandomSlotItem() {
    return Math.floor(Math.random() * ( constants.numberOfSlotElement - 1 + 1)) + 1;
}

describe("tests for slot container component", () => {
    it("Should load properly for for initial load", ()=>{
        const { wrapper } = setup({
            rollerData: null,
            rollers: [],
            spinningFlag: false,
            waitToSpinChannel: null,
            spinChannel: null,
            stopChannel: null,
            result: false
        });
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for initial load");
    });
    it("Should load properly for rollers(wheels) data on page load", () => {
        const { wrapper } = setup({
            rollers: [  rollerData[generateRandomSlotItem()-1],
                        rollerData[generateRandomSlotItem()-1],
                        rollerData[generateRandomSlotItem()-1]
                    ],
            spinning: false,
            result: false
        });
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for rollers(wheels) data on page load");
    });
    it("Should load properly for rollers(wheels) spinning", () => {
        const { wrapper } = setup({
            rollers: [  rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1]
            ],
            spinning: true,
            result: false
        });
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for rollers(wheels) spinning");
    });
    it("Should load properly for showing result", () => {
        const { wrapper } = setup({
            rollers: [  rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1]
            ],
            spinning: false,
            result: "you won $20"
        });
        expect(shallowToJson(wrapper)).toMatchSnapshot("test for showing result");
    });
    it('should call method passed to OnstartSpin on clicking the start button', ()=> {
        const { wrapper } = setup({
            rollers: [  rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1]
            ],
            spinning: false,
            result: true,
            onStart: mockOnStartSpin
        });
        wrapper.find("#startBtn").simulate('click');
        expect(mockOnStartSpin).toHaveBeenCalledWith();
    });
    it('should call method passed to OnstopSpin on clicking the stop button', ()=> {
        const { wrapper } = setup({
            rollers: [  rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1],
                rollerData[generateRandomSlotItem()-1]
            ],
            spinning: true,
            result: false,
            onStop: mockOnStopSpin
        });
        wrapper.find("#stopBtn").simulate('click');
        expect(mockOnStopSpin).toHaveBeenCalledWith();
    });
});