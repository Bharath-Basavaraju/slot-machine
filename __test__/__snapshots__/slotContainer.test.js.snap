// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for slot container component Should load properly for for initial load: test for initial load 1`] = `
<div>
  <div />
  <div>
    <button
      className="undefined "
      id="startBtn"
    >
      Start
    </button>
    <button
      className="undefined undefined"
      disabled={true}
      id="stopBtn"
    >
      Stop
    </button>
  </div>
</div>
`;

exports[`tests for slot container component Should load properly for rollers(wheels) data on page load: test for rollers(wheels) data on page load 1`] = `
<div>
  <div>
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -0",
          "id": "strawberry",
        }
      }
      key="0"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -0",
          "id": "strawberry",
        }
      }
      key="1"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -200px",
          "id": "orange",
        }
      }
      key="2"
    />
  </div>
  <div>
    <button
      className="undefined "
      disabled={false}
      id="startBtn"
    >
      Start
    </button>
    <button
      className="undefined undefined"
      disabled={true}
      id="stopBtn"
    >
      Stop
    </button>
  </div>
</div>
`;

exports[`tests for slot container component Should load properly for rollers(wheels) spinning: test for rollers(wheels) spinning 1`] = `
<div>
  <div>
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -0",
          "id": "strawberry",
        }
      }
      key="0"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -300px",
          "id": "monkey",
        }
      }
      key="1"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -200px",
          "id": "orange",
        }
      }
      key="2"
    />
  </div>
  <div>
    <button
      className="undefined undefined"
      disabled={true}
      id="startBtn"
    >
      Start
    </button>
    <button
      className="undefined "
      disabled={false}
      id="stopBtn"
    >
      Stop
    </button>
  </div>
</div>
`;

exports[`tests for slot container component Should load properly for showing result: test for showing result 1`] = `
<div>
  <div>
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -100px",
          "id": "banana",
        }
      }
      key="0"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -100px",
          "id": "banana",
        }
      }
      key="1"
    />
    <RollerComponent
      elementData={
        Object {
          "backgroundPosition": "-0 -300px",
          "id": "monkey",
        }
      }
      key="2"
    />
  </div>
  <div>
    <button
      className="undefined "
      disabled={false}
      id="startBtn"
    >
      Start
    </button>
    <button
      className="undefined undefined"
      disabled={true}
      id="stopBtn"
    >
      Stop
    </button>
  </div>
  <div>
    you won $20
  </div>
</div>
`;
